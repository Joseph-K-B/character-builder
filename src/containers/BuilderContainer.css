html{
    background-color: black;
}

main{
    background-color: #272838;
    display: grid;   
    grid-template-columns: 1fr 2fr 1fr;
}

footer{
    font-size: 10%;
    color: white;
}
