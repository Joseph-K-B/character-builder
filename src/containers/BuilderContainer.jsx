import React, { useState } from "react";
import Character from "../components/Character/Character";
import Picker from "../components/Picker/Picker";
import Stats from "../components/Stat/Stats";

function BuilderContainer() {
    const [head, setHead] = useState('dog')
    const [torso, setTorso] = useState('leather')
    const [bottom, setBottom] = useState('jeans')
    const [catchPhrase, setCatchPhrase] = useState('')
    const [catchPhraseArr, setCatchPhraseArr] = useState([])

    // const onButtonClick = () => {
    //     console.log('BUTTON')
    // }

    return (
        <>
        <main>
            <Picker
            head={head}
            setHead={setHead}
            torso={torso}
            setTorso={setTorso}
            bottom={bottom}
            setBottom={setBottom} />
            <Character
            head={head}
            setHead={setHead}
            torso={torso}
            setTorso={setTorso}
            bottom={bottom}
            setBottom={setBottom} />
            <Stats 
                catchPhrase={catchPhrase}
                setCatchphrase={setCatchPhrase}
            />
        </main>
        </>
    )
}

export default BuilderContainer;