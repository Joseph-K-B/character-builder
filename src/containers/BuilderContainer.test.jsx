import { render, screen } from "@testing-library/react";
import BuilderContainer from "./BuilderContainer";

it('should render container with all presentational components', () => {
    const {container} = render(<BuilderContainer catchPhraseArr={['hello', 'goodbye']}/>);

    const selectHead = screen.getByLabelText(/Head/);
    const selectTorso = screen.getByLabelText(/Torso/);
    const selectBottom = screen.getByLabelText(/Bottom/);
    const headImg = screen.getByLabelText(/head/);
    const torsoImg = screen.getByLabelText(/torso/);
    const bottomImg = screen.getByLabelText(/bottom/);
    const input = screen.getByLabelText('catchphrase-input');
    const button = screen.getByRole('button');
    const changeCount = screen.getByLabelText('change-count')
    const list = screen.getByLabelText('catch-list');

    expect(headImg).toBeInTheDocument();
    expect(torsoImg).toBeInTheDocument();
    expect(bottomImg).toBeInTheDocument();
    expect(selectHead).toBeInTheDocument();
    expect(selectTorso).toBeInTheDocument();
    expect(selectBottom).toBeInTheDocument();
    expect(changeCount).toBeInTheDocument();
    expect(list).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})