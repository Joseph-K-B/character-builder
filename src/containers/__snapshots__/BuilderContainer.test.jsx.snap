// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render container with all presentational components 1`] = `
<div>
  <main>
    <section
      class="image-controls"
    >
      <div>
        <label
          for="head"
        >
          Head
        </label>
        <select
          id="head"
          name="head"
        >
          <option>
            dog
          </option>
          <option>
            dog-2
          </option>
          <option>
            dog-3
          </option>
          <option>
            duck
          </option>
          <option>
            horse
          </option>
          <option>
            cat
          </option>
        </select>
      </div>
      <div>
        <label
          for="torso"
        >
          Torso
        </label>
        <select
          id="torso"
          name="torso"
        >
          <option>
            leather
          </option>
          <option>
            suit
          </option>
          <option>
            pink
          </option>
        </select>
      </div>
      <div>
        <label
          for="torso"
        >
          Bottom
        </label>
        <select
          id="bottom"
          name="bottom"
        >
          <option>
            jeans
          </option>
          <option>
            white
          </option>
          <option>
            blue
          </option>
        </select>
      </div>
    </section>
    <section
      class="image-section"
    >
      <figure
        aria-label="head"
        class="head-image"
        data-testid="head"
        style="background-image: url(./dog-head.png);"
        value="dog"
      />
      <figure
        aria-label="torso"
        class="torso-image"
        data-testid="torso"
        style="background-image: url(./leather-torso.png);"
        value="leather"
      />
      <figure
        aria-label="bottom"
        class="bottom-image"
        data-testid="bottom"
        style="background-image: url(./jeans-bottom.png);"
        value="jeans"
      />
    </section>
    <section
      class="stat-controls"
    >
      <input
        aria-label="catchphrase-input"
        data-testid="catchphrase-input"
        id="catchphrase"
        name="catchphrase"
        type="text"
        value=""
      />
      <label
        for="catchphrase"
      />
      <button
        type="button"
      >
        Submit
      </button>
    </section>
    <div>
      <p
        aria-label="change-count"
      >
        The head of your charachter has been updated 
        0
         times, the torso updated 
        0
        , and the legs updated 
        0
         times.
      </p>
      <ul
        aria-label="catch-list"
      >
        Catchphrases
      </ul>
    </div>
  </main>
</div>
`;
